applications/nrf5340_audio/src/utils/macros/macros_common.h

```diff
@@ -0,0 +1,112 @@
+ */
+
+#if defined(CONFIG_DEBUG)
+#define PRINT_AND_OOPS(code)                                                                       \
+	do {                                                                                       \
+		LOG_ERR("ERR_CHK Err_code: [%d] @ line: %d\t", code, __LINE__);                    \
+		k_oops();                                                                          \
+	} while (0)
+#else
+#define PRINT_AND_OOPS(code)                                                                       \
+	do {                                                                                       \
+		LOG_ERR("ERR_CHK Err_code: [%s] @ line: %d\t", STRINGIFY(code), __LINE__);         \
+		k_oops();                                                                          \
+	} while (0)
+#endif /* (CONFIG_DEBUG) */
+
+#define ERR_CHK(err_code)                                                                          \
+	do {                                                                                       \
+		if (err_code) {                                                                    \
+			PRINT_AND_OOPS(err_code);                                                  \
+		}                                                                                  \
+	} while (0)
+
+#define ERR_CHK_MSG(err_code, msg)                                                                 \
+	do {                                                                                       \
+		if (err_code) {                                                                    \
+			LOG_ERR("%s", msg);                                                        \
+			PRINT_AND_OOPS(err_code);                                                  \
+		}                                                                                  \
+	} while (0)
+
+#define ERR_ASSERT(assert_val)                                                                     \
+	do {                                                                                       \
+		if (!(assert_val)) {                                                               \
+			PRINT_AND_OOPS(assert_val);                                                \
+		}                                                                                  \
+	} while (0)
+
+#define ERR_ASSERT_MSG(assert_val, msg)                                                            \
+	do {                                                                                       \
+		if (!(assert_val)) {                                                               \
+			LOG_ERR("%s", msg);                                                        \
+			PRINT_AND_OOPS(assert_val);                                                \
+		}                                                                                  \
+	} while (0)
+
+#define RET_IF_ERR(err_code)                                                                       \
```

Comment:
@cvinayak @rugeGerritsen @gmarull @carlescufi : We were made aware of the CHECKIF macro in Zephyr which solves the ERR_ASSERT macros here. We will submit a PR to expand this to include a message/string.

What remains to be solved is this RET_IF_ERR. Do you know of any similar functionality we can use in Zephyr? - -- -Or how we can handle without adding `if` everywhere in the code?

Reply:
@koffes Zephyr (and then I suppose by extension also NCS) follow the Linux Kernel style guide (with some exceptions). 

In the Linux Kernel Style guide it is mentioned that macros that affect flow control should be avoided: https://www.kernel.org/doc/html/v4.10/process/coding-style.html#macros-enums-and-rtl

That means that we should not have any macro that returns from a function, and thus we can't really avoid `if` to replace this one.